Model/StoreIntegrationRepository.php

```diff
@@ -99,10 +99,15 @@ public function getByStoreIdAndActiveEnvironment(int $storeId): StoreIntegration
     public function getByStoreIdAndIntegrationId(int $storeId, int $integrationId): StoreIntegrationInterface
     {
         $storeIntegrationCollection = $this->storeIntegrationCollectionFactory->create();
-        return $storeIntegrationCollection
+        $storeIntegration = $storeIntegrationCollection
             ->addFieldToFilter('store_id', ['eq' => $storeId])
             ->addFieldToFilter('integration_id', ['eq' => $integrationId])
             ->getFirstItem();
```

Comment:
`getFirstItem` returns an empty object if the record can not be located which caused the API to return a 200 with an empty array